src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java

```diff
@@ -5,48 +5,75 @@
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.AnalogGyro;
 import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.RobotController;
 import edu.wpi.first.wpilibj.drive.DifferentialDrive;
-import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.simulation.AnalogGyroSim;
+import edu.wpi.first.wpilibj.simulation.DifferentialDrivetrainSim;
+import edu.wpi.first.wpilibj.simulation.EncoderSim;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.ADIS16448_IMU;
 import edu.wpi.first.math.controller.SimpleMotorFeedforward;
-import edu.wpi.first.math.geometry.Pose2d;
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.RelativeEncoder;
 
 import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj.I2C;
 import com.kauailabs.navx.frc.AHRS;
 
-import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 public class DriveBaseSubsystem extends SubsystemBase {
 
-  private Joystick m_driverJoystick;
-  private MotorController[] m_motorControllers;
-  private DifferentialDrive m_differentialDrive;
+  private final Joystick m_driverJoystick;
+  private final MotorController[] m_motorControllers;
+  private final DifferentialDrive m_differentialDrive;
+  private DifferentialDrivetrainSim m_DifferentialDrivetrainSim;
+  public final Field2d m_field = new Field2d();
   private AHRS m_gyro;
-  private DifferentialDriveOdometry m_odometry;
-
+  private AnalogGyroSim m_gyroSim;
+  public static ADIS16448_IMU m_gyro2; //Non-native gyro, might use later
+  private AnalogGyro m_gyro1;
+  private final DifferentialDriveOdometry m_odometry;
+  public static Encoder m_leftEncoder;
+  public static Encoder m_rightEncoder;
+  private EncoderSim m_leftEncoderSim;
+  private EncoderSim m_rightEncoderSim;
+
+  //Encoders for Sim
+  private Encoder m_LEncoderForSim;
+  private Encoder m_REncoderForSim;
+ 
   // external encoders
   private Encoder m_extRightEncoder;
   private Encoder m_extLeftEncoder;
 
   // internal encoders
   private RelativeEncoder m_internalLeftEncoder;
   private RelativeEncoder m_internalRightEncoder;
-
   private SimpleMotorFeedforward m_sMotorFeedforward;
 
   private boolean usingExternal = false;
 
   
   public DriveBaseSubsystem(Joystick joystick, boolean usingExternal) {  
     m_driverJoystick = joystick;
+
     m_motorControllers = new MotorController[4];
     m_gyro = new AHRS(I2C.Port.kMXP);
     m_gyro.reset(); // resets the heading of the robot to 0
+    m_gyro1 = new AnalogGyro(1);
+    m_LEncoderForSim = new Encoder(0, 1);
+    m_REncoderForSim = new Encoder(2, 3);
```

Comment:
Will conflict with external encoder channels if usingExternal is ever set to True. We should look for a way to only create the encoders we need for the given setup (simulation, real internal encoder, real external encoder) so they don't collide.

Reply:
Also, would be nice to document why these values were chosen (either make them Constants variables or put a comment)

Reply:
I put both of them in two if statements that made them meet the requirements that you said. Not 100% sure if that is what you meant but I think it works